src/url-loader/url-resolver.ts

```diff
@@ -12,13 +12,16 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {posix} from 'path';
-import {resolve as urlLibResolver} from 'url';
+import * as path from 'path';
+import {format as urlLibFormat, resolve as urlLibResolver} from 'url';
 
 import {parseUrl} from '../core/utils';
 import {PackageRelativeUrl, ScannedImport} from '../index';
 import {FileRelativeUrl, ResolvedUrl} from '../model/url';
 
+const sharedRelativeUrlProperties =
+    ['protocol', 'slashes', 'auth', 'host', 'port', 'hostname'];
```

Comment:
Can avoid the `as any` business below by writing this as:

```typescript
['protocol' as 'protocol', 'slashes' as 'slashes', 'auth' as 'auth', 
 'host' as 'host', 'port' as 'port', 'hostname' as 'hostname']
```

Reply:
I am both excited to learn this technique and yet bothered by it all the same; but I totally get it.  Thanks!

Reply:
I went ahead and dropped the array anyways since I the logic for checking each of the four properties was at least as easy to follow as iterating.